exaudfclient/docs/script_options_requirments.md

```diff
@@ -0,0 +1,376 @@
+
+Needs: req
+
+## High-level Requirements
+
+This section details the high-level requirements for the new parser system, linked to the features listed above.
+
+### General Script Options Parsing
+`req~general-script-options-parsing~1`
+
+The parser must correctly identify and handle Script Options with the syntax `%<optionKey><white spaces><optionValue>;`.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+### White Spaces
+`req~white-spaces~1`
+
+The parser must treat the following list of white spaces as token separator:
+|======================================================= 
+| Name         | C syntax      | ASCII Dec | ASCII Hex | 
+| tabulator    | '\t'          | 9         | 0x09      |
+| vertical tab | '\v'          | 11         | 0x0b     |
+| form feed    | '\f'          | 12         | 0x0c     |
+| space        | ' '           | 30        | 0x20      |     
+|======================================================= 
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+
+### White Spaces Separator
+`req~white-spaces-separator~1`
+
+A Script Options can have one or more white spaces between the Script Option Key `<optionKey>` and Script Option Value `<<optionValue>`.
+The parser must recognize those white spaces as separator(s).
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### Leading White Spaces Options Parsing
+`req~leading-white-spaces-script-options-parsing~1`
+
+The parser must accept Script Options for lines starting with any number of white space characters before the Script Options.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### Ignore anything which is not a Script Option
+`req~ignore-none-script-options~1`
+
+If there is any character in front of a Script Option which is not a white space, the parser must ignore the option(s). 
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### Multiple Line Script Options Parsing
+`req~multiple-lines-script-options-parsing~1`
+
+The parser must recognize Script Options at any line in the given script code.
+
+Needs: dsn
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+### White Spaces Options Parsing V1
+`req~white-spaces-script-options-parsing-v1~1`
+
+All white spaces between the option key and option value are to be interpreted as separator. 
+White spaces between the option value and the terminating ";" are to be removed from the option value.
+
+Needs: dsn
+
+Tags: V1
+
+Covers:
+- `feat~general-script-options-parsing~1`
+
+Depends:
+ - `req~white-spaces~1`
+
+### White Spaces Options Parsing V2
+`req~white-spaces-script-options-parsing-v2~1`
+
+All white spaces between the option key and option value are to be ignored. The following rules for escape sequences at **the start** of a script optionValue are to be applied:
```

Comment:
Its actually not only at the start I guess, but we need it at the start

Reply:
Escape sequences which are not at the start are not converted.
Compare https://github.com/exasol/script-languages/blob/f2ba8d2b531ecde327ca11f84f50c31594c6f80d/exaudfclient/base/script_options_parser/ctpg/script_option_lines_ctpg.cc#L146 versus https://github.com/exasol/script-languages/blob/f2ba8d2b531ecde327ca11f84f50c31594c6f80d/exaudfclient/base/script_options_parser/ctpg/script_option_lines_ctpg.cc#L140

Reply:
add rationale, we wanted to stay as much as possible backwards compatible

Reply:
Done